File: uri/strings/imagen/1516.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

int main() {
    int n, m;
    
    while (cin >> n >> m && (n != 0 || m != 0)) {
        vector<string> imagen(n);
        
        for (int i = 0; i < n; i++) {
            cin >> imagen[i];
        }
        
        int a, b;
        cin >> a >> b;
        
        int factorV = a / n;
        int factorH = b / m;
        
        for (int i = 0; i < n; i++) {
            string lineaExpandida = "";
            for (int j = 0; j < m; j++) {
                for (int k = 0; k < factorH; k++) {
                    lineaExpandida += imagen[i][j];
                }
            }
            
            for (int k = 0; k < factorV; k++) {
                cout << lineaExpandida << endl;
            }
        }
        
        cout << endl;
    }
    
    return 0;
}

File: uri/strings/imagen/1516.rs
use std::io;

fn main() {
    loop {
        let mut input = String::new();
        io::stdin().read_line(&mut input).unwrap();
        let dims: Vec<usize> = input
            .trim()
            .split_whitespace()
            .map(|x| x.parse().unwrap())
            .collect();
        
        let n = dims[0];
        let m = dims[1];
        
        if n == 0 && m == 0 {
            break;
        }
        
        let mut imagen: Vec<String> = Vec::new();
        for _ in 0..n {
            let mut linea = String::new();
            io::stdin().read_line(&mut linea).unwrap();
            imagen.push(linea.trim_end().to_string());
        }
        
        let mut input = String::new();
        io::stdin().read_line(&mut input).unwrap();
        let nuevas_dims: Vec<usize> = input
            .trim()
            .split_whitespace()
            .map(|x| x.parse().unwrap())
            .collect();
        
        let a = nuevas_dims[0];
        let b = nuevas_dims[1];
        
        let factor_v = a / n;
        let factor_h = b / m;
        
        for i in 0..n {
            let mut linea_expandida = String::new();
            for ch in imagen[i].chars() {
                for _ in 0..factor_h {
                    linea_expandida.push(ch);
                }
            }
            
            for _ in 0..factor_v {
                println!("{}", linea_expandida);
            }
        }
        
        println!();
    }
}
